### PR TITLE
Adding unittests for the sha224 module

### PR DESCRIPTION
**Pull request by jebrando**

### CODE REVIEW

File: tests/sha_ut/sha_ut.c [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+    TEST_SUITE_INITIALIZE(TestClassInitialize)
+    {
+        g_testByTest = TEST_MUTEX_CREATE();
+        ASSERT_IS_NOT_NULL(g_testByTest);
+
+        (void)umock_c_init(on_umock_c_error);
+    }
+
+    /**
+     * The test suite will call this function to cleanup your machine.
```
Comment (dcristoloveanu) [inline tests/sha_ut/sha_ut.c]: >cleanup your machine. [](start = 49, length = 21)

Lies! It's not cleaning up the machine, do not rely on this!

File: tests/sha_ut/sha_ut.c [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+
+        umock_c_reset_all_calls();
+    }
+
+    TEST_FUNCTION_CLEANUP(cleans)
+    {
+        TEST_MUTEX_RELEASE(g_testByTest);
+    }
+
+    TEST_FUNCTION(SHA224Reset_succeed)
```
Comment (dcristoloveanu) [inline tests/sha_ut/sha_ut.c]: >succeed [](start = 30, length = 7)

succeeds

Comment (dcristoloveanu) [reply tests/sha_ut/sha_ut.c]: For the other tests too

---
In reply to: [237965997](https://github.com/Azure/azure-c-shared-utility/pull/236#discussion_r237965997) [](ancestors = 237965997)

File: tests/sha_ut/sha_ut.c [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+        result = SHA256Result(&sha_ctx, Message_Digest);
+
+        //assert
+        ASSERT_ARE_EQUAL(int, 0, result);
+        ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+        //cleanup
+    }
+
+    TEST_FUNCTION(SHA224Result_succeed)
```
Comment (dcristoloveanu) [inline tests/sha_ut/sha_ut.c]: Nice to see tests being added!

File: src/sha224.c [hunk 4]
```diff
@@ -294,24 +292,28 @@ int SHA256FinalBits(SHA256Context *context,
...
-        return context->Corrupted;
-
-    SHA224_256AddLength(context, length);
-    SHA224_256Finalize(context, (uint8_t)
-        ((message_bits & masks[length]) | markbit[length]));
-
-    return shaSuccess;
+    else if (!context)
+    {
+        result = shaNull;
```
[the remark this answers is not in the document]
Comment (jebrando) [reply src/sha224.c]: Did not like that we were returning in the middle of the function.  Just to make it consistent with the rest of the repo.

---
In reply to: [237966941](https://github.com/Azure/azure-c-shared-utility/pull/236#discussion_r237966941) [](ancestors = 237966941)

**Pull request merged**